### PR TITLE
separate graph and persistedGraph

### PR DESCRIPTION
**Pull request by jim-minter**
### Which issue this PR addresses:

Should help some production admin upgrade failures on old clusters.

### What this PR does / why we need it:

We are running into trouble where we can't straightforwardly unmarshal old
cluster graphs (the serialisation of some ignition structs has changed enough
to break unmarshalling).  Fundamentally the issue is that the installer does
not intend for the graph to be a stable persisted API, which is fair enough,
but it makes this hard to fix.

Separate graph and persistentGraph:
* graph is a one-time object: create it, resolve it, persist it, done.
* persistedGraph can be read subsequently and, with care, potentially
  round-tripped if necessary.

Callers of persistentGraph.get() receive any errors when unmarshalling the
object that interests them, so they are forced to handle it.

Using json.RawMessage also has the side effect that we shouldn't be at as much
risk of accidentally losing data if we ever needed to round-trip the graph.

### Test plan for issue:

E2E

### Is there any documentation that needs to be updated for this PR?

No.

### CODE REVIEW

File: pkg/cluster/graph_test.go [hunk 1]
```diff
@@ -1,164 +0,0 @@
...
-	"golang.org/x/crypto/ssh"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
-
-// TestGraphRoundTrip builds a representative graph, then marshals and
-// unmarshals it.  It tests that registeredTypes isn't missing any obvious keys,
-// and along the way is a smoke test for graph.resolve, ensuring (among other
-// things) that it does not attempt to read values from stdin and that it can
-// find its assets.
-func TestGraphRoundTrip(t *testing.T) {
```
Comment (jim-minter) [inline pkg/cluster/graph_test.go:33]: E2E covers this.

File: pkg/cluster/graph.go [hunk 2]
```diff
@@ -7,151 +7,26 @@ import (
...
-	"github.com/openshift/installer/pkg/asset/openshiftinstall"
-	"github.com/openshift/installer/pkg/asset/password"
-	"github.com/openshift/installer/pkg/asset/releaseimage"
-	"github.com/openshift/installer/pkg/asset/rhcos"
-	"github.com/openshift/installer/pkg/asset/templates/content/bootkube"
-	"github.com/openshift/installer/pkg/asset/templates/content/openshift"
-	"github.com/openshift/installer/pkg/asset/tls"
 )
 
-var registeredTypes = map[string]asset.Asset{
```
Comment (jim-minter) [inline pkg/cluster/graph.go:32]: :tada:

**Comment by jim-minter** [comment]
@nilsanderselde if you have a chance, ptal at this

**Review by nilsanderselde — APPROVED** [review]
LGTM. Cleans things up a bit, too, which is nice.

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**